src/test/java/org/mockito/internal/creation/instance/ConstructorInstantiatorTest.java

```diff
@@ -46,7 +46,7 @@ public void explains_when_constructor_cannot_be_found() {
             fail();
         } catch (InstantiationException e) {
             assertThat(e).hasMessageContaining("Unable to create instance of 'SomeClass2'.\n" +
-                    "Please ensure it has 0-arg constructor which invokes cleanly.");
+                    "Please ensure that it has a constructor that matches these arguments: []");
```

Comment:
With this change there's no longer any distinction between a 0-arg constructor, a 1-arg constructor or a 12-arg constructor, so I'm not sure I can retain this exception messages. 

Perhaps the 0-arg constructor can be singled out in `ConstructorInstantiator#paramsException(Class ,Exception)`, but that's probably the only one. Would you recomend that?